data.sql

```diff
@@ -55,3 +55,47 @@ WHERE a.name = 'Agumon' AND o.full_name = 'Sam Smith'
   OR a.name IN ('Devimon', 'Plantmon') AND o.full_name = 'Bob'
   OR a.name IN ('Charmander', 'Squirtle', 'Blossom') AND o.full_name = 'Melody Pond'
   OR a.name IN ('Angemon', 'Boarmon') AND o.full_name = 'Dean Winchester';
+
+
+-- Insert data for vets
+INSERT INTO vets (name, age, date_of_graduation)
+VALUES
+  ('William Tatcher', 45, '2000-04-23'),
+  ('Maisy Smith', 26, '2019-01-17'),
+  ('Stephanie Mendez', 64, '1981-05-04'),
+  ('Jack Harkness', 38, '2008-06-08');
+
+-- Insert data for vet specializations
+INSERT INTO specializations (vet_id, species_id)
+SELECT v.id AS vet_id, s.id AS species_id FROM vets AS v
+JOIN species AS s ON v.name = 'William Tatcher' AND s.name = 'Pokemon'
+UNION
+SELECT v.id AS vet_id, s.id AS species_id FROM vets AS v
+JOIN species AS s ON v.name = 'Stephanie Mendez' AND (s.name = 'Digimon' OR s.name = 'Pokemon')
+UNION
+SELECT v.id AS vet_id, s.id AS species_id FROM vets AS v
+JOIN species AS s ON v.name = 'Jack Harkness' AND s.name = 'Digimon';
+
+-- Insert data for visits
+INSERT INTO visits (animal_id, vet_id, visit_date)
+VALUES
+  ((SELECT id FROM animals WHERE name = 'Agumon'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2020-05-24'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Agumon'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2020-07-22'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Gabumon'), (SELECT id FROM vets WHERE name = 'Jack Harkness'), '2021-02-02'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Pikachu'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2020-01-05'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Pikachu'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2020-03-08'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Pikachu'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2020-05-14'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Devimon'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2021-05-04'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Charmander'), (SELECT id FROM vets WHERE name = 'Jack Harkness'), '2021-02-24'::DATE),
+  ((SELECT id FROM animals WHERE name = 'Plantmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-12-21'::DATE),
```

Comment:
- [ ] When i try using these lines of code to insert data for your visits table, i get the below error and as such i am unable to test your queries to see that they are working, please kindly use the code snippet below as a guide to add data into your database :+1: 
![rer](https://github.com/lily4178993/VetClinicDB/assets/89055502/66139ba2-c4e5-44d7-97e7-d13c78bcd89a)
- code snippet
 ```sql
INSERT INTO visits (vet_id, animal_id, visit_date)
VALUES
(1, 4, '2020-05-24'),
(1, 3, '2020-07-22'),
(4, 5, '2021-02-02'),
(2, 10, '2020-01-05'),
(2, 10, '2020-03-08'),
(2, 10, '2020-05-14'),
(3, 6, '2021-05-04'),
(4, 7, '2021-02-24'),
(2, 1, '2019-12-21'),
(1, 1, '2020-08-10'),
(2, 1, '2021-04-07'),
(3, 8, '2019-09-29'),
(4, 2, '2020-10-03'),
(4, 2, '2020-11-04'),
(2, 3, '2019-01-24'),
(2, 3, '2019-05-15'),
(2, 3, '2020-02-27'),
(2, 3, '2020-08-03'),
(3, 9, '2020-05-24'),
(1, 9, '2021-01-11');
```

Reply:
Hello @Efiamotu-1,

Thank you for your review. I appreciate your feedback. The provided code is functioning correctly, and I was able to successfully add data using it.

Reply:
If you could provide more specific details regarding the problem you've encountered, it would be greatly appreciated.

Reply:
It seems that using this code snippet would result in hardcoded entries. This could lead to a potential issue when trying to match IDs. If you have any alternative suggestions or ideas to make the code more dynamic and less reliant on hardcoding, I'd greatly appreciate your insights.

Reply:
I had been eagerly awaiting your insights for several hours. The issue with hardcoding IDs in the 'Join Table' visits has been concerning, as the order of data insertion could affect the IDs. Your input is invaluable in addressing this concern.